### PR TITLE
mamali/check-for-mt5-api-error

### PR DESCRIPTION
**Pull request by reza-fs**

### CODE REVIEW

File: src/javascript/_common/base/client_base.js [hunk 1]
```diff
@@ -316,7 +316,7 @@ const ClientBase = (() => {
 
     const canChangeCurrency = (statement, mt5_login_list, is_current = true) => {
         const currency             = get('currency');
-        const has_no_mt5           = mt5_login_list.length === 0;
+        const has_no_mt5           = mt5_login_list ? mt5_login_list.length === 0 : true;
```
Comment (negarn) [inline src/javascript/_common/base/client_base.js]: can we change this to:
```suggestion
        const has_no_mt5           = !mt5_login_list || !mt5_login_list.length;
```
I think it's less confusing

File: src/javascript/app/pages/user/metatrader/metatrader.js [hunk 2]
```diff
@@ -22,7 +22,12 @@ const MetaTrader = (() => {
         BinarySocket.wait('landing_company', 'get_account_status', 'statement').then(async () => {
             if (isEligible()) {
                 if (Client.get('is_virtual')) {
-                    await addAllAccounts();
+                    try {
+                        await addAllAccounts();
+                    } catch (error) {
+                        // eslint-disable-next-line no-console
+                        console.warn(error.message);
```
Comment (negarn) [inline src/javascript/app/pages/user/metatrader/metatrader.js]: why are we just warning in console?

Comment (reza-fs) [reply src/javascript/app/pages/user/metatrader/metatrader.js]: I wanted to ignore it, but I thought I left a trace for further debugging. I can just do nothing in the error block, what do you suggest?

File: src/javascript/_common/base/client_base.js (hunk 1 above)
Comment (reza-fs) [reply src/javascript/_common/base/client_base.js]: Sure. Will do.

File: src/javascript/app/pages/user/metatrader/metatrader.js [hunk 3]
```diff
@@ -68,8 +66,12 @@ const MetaTrader = (() => {
     };
 
     const addAllAccounts = () => (
-        new Promise((resolve) => {
+        new Promise((resolve, reject) => {
             BinarySocket.wait('mt5_login_list').then((response) => {
+                if (response.error) {
+                    reject(response.error);
```
Comment (negarn) [inline src/javascript/app/pages/user/metatrader/metatrader.js]: I think we can't proceed if there's an error; user can't really use the dashboard unless we know what their MT5 accounts are. I suggest using the error function of metatrader to display that error message and ask user to try again.

Comment (reza-fs) [reply src/javascript/app/pages/user/metatrader/metatrader.js]: Yeah, that is a great idea, will work on it.

File: src/javascript/app/pages/user/metatrader/metatrader.js [hunk 4]
```diff
@@ -22,7 +22,11 @@ const MetaTrader = (() => {
         BinarySocket.wait('landing_company', 'get_account_status', 'statement').then(async () => {
             if (isEligible()) {
                 if (Client.get('is_virtual')) {
-                    await addAllAccounts();
+                    try {
+                        await addAllAccounts();
```
Comment (negarn) [inline src/javascript/app/pages/user/metatrader/metatrader.js]: how about the second usage of this on line 33?

Comment (reza-fs) [reply src/javascript/app/pages/user/metatrader/metatrader.js]: Done.

Comment (negarn) [reply src/javascript/app/pages/user/metatrader/metatrader.js]: what is done? 😅 I see no change committed

**Pull request merged**